sklearn/frozen/tests/test_frozen.py

```diff
@@ -0,0 +1,215 @@
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+import re
+
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+
+from sklearn.base import (
+    BaseEstimator,
+    clone,
+    is_classifier,
+    is_clusterer,
+    is_outlier_detector,
+    is_regressor,
+)
+from sklearn.cluster import KMeans
+from sklearn.compose import make_column_transformer
+from sklearn.datasets import make_classification, make_regression
+from sklearn.exceptions import NotFittedError, UnsetMetadataPassedError
+from sklearn.frozen import FrozenEstimator
+from sklearn.linear_model import LinearRegression, LogisticRegression
+from sklearn.neighbors import LocalOutlierFactor
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import RobustScaler, StandardScaler
+from sklearn.utils.validation import check_is_fitted
+
+REGRESSION_DATASET = make_regression()
+CLASSIFICATION_DATASET = make_classification()
+
+
+@pytest.mark.parametrize(
+    "estimator, dataset",
+    [
+        (LinearRegression(), REGRESSION_DATASET),
+        (LogisticRegression(), CLASSIFICATION_DATASET),
+        (make_pipeline(StandardScaler(), LinearRegression()), REGRESSION_DATASET),
+        (make_pipeline(StandardScaler(), LogisticRegression()), CLASSIFICATION_DATASET),
+        (StandardScaler(), REGRESSION_DATASET),
+        (KMeans(), REGRESSION_DATASET),
+        (LocalOutlierFactor(), REGRESSION_DATASET),
+        (
+            make_column_transformer(
+                (StandardScaler(), [0]),
+                (RobustScaler(), [1]),
+            ),
+            REGRESSION_DATASET,
+        ),
+    ],
+)
+def test_frozen_methods(estimator, dataset):
+    """Test that frozen.fit doesn't do anything, and that all other methods are
+    exposed by the frozen estimator and return the same values as the estimator.
+    """
+    X, y = dataset
+    estimator.fit(X, y)
+    frozen = FrozenEstimator(estimator)
+    # this should be no-op
+    frozen.fit([[1]], [1])
+
+    methods = [
+        "predict",
+        "predict_proba",
+        "predict_log_proba",
+        "decision_function",
+        "transform",
+    ]
+    for method in methods:
```

Comment:
We could parametrize this one as well. This could be handy to know which method fail directly and we could inside the test have a `skip` if does not apply.